components/AppContext.tsx

```diff
@@ -1,10 +1,15 @@
 import React, { createContext, useContext, useState, useEffect } from "react";
 import AsyncStorage from "@react-native-async-storage/async-storage";
+import { BasicAuthInformation } from "../interfaces/basic-auth-information";
 
 // Create a context
 const AppContext = createContext({
   serverUrl: "",
-  updateServerUrl: async (url: string) => {
+  basicAuthInformation: { basicAuthRequired: false } as BasicAuthInformation,
```

Comment:
```suggestion
  basicAuth: { required: false } as BasicAuth,
```

would keep it shorter